src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/scheduler/BlobProcessor.java

```diff
@@ -0,0 +1,143 @@
+import java.util.zip.ZipInputStream;
+
+import static com.google.common.io.ByteStreams.toByteArray;
+
+/**
+ * This class is a scheduler which runs as per configured interval.
+ * It will read all the blobs from Azure Blob storage and will do below things
+ * 1. Reads Blob from container.
+ * 2. Extract Zip file(Blob)
+ * 3. Transform metadata json to DB entities.
+ * 4. Save PDF files in document storage.
+ * 5. Update status and doc urls in DB.
+ */
+@Component
+@ConditionalOnProperty(value = "scheduling.enabled", matchIfMissing = true)
+public class BlobProcessor {
+
+    private static final Logger log = LoggerFactory.getLogger(BlobProcessor.class);
+
+    private final CloudBlobClient cloudBlobClient;
+    private final EnvelopeRepository envelopeRepository;
+    private final DocumentManagementService documentManagementService;
+
+    @Autowired
+    public BlobProcessor(
+        CloudBlobClient cloudBlobClient,
+        EnvelopeRepository envelopeRepository,
+        DocumentManagementService documentManagementService
+    ) {
+        this.cloudBlobClient = cloudBlobClient;
+        this.envelopeRepository = envelopeRepository;
+        this.documentManagementService = documentManagementService;
+    }
+
+    @Scheduled(fixedDelayString = "${scan.delay}")
+    public void processBlobs() {
+        cloudBlobClient.listContainers()
+            .forEach(cloudBlobContainer ->
+                processBlobsPerContainer(cloudBlobContainer.getName())
+            );
+    }
+
+    private void processBlobsPerContainer(String containerName) {
+        log.info("Processing blobs for container {} ", containerName);
+
+        try {
+            processZipFiles(cloudBlobClient.getContainerReference(containerName));
+        } catch (URISyntaxException | StorageException e) {
+            log.error("Exception occurred while retrieving container reference for " + containerName, e);
+            // TODO Raise an event in DB or track in appinsights if this happens.
+        }
+    }
+
+    private void processZipFiles(CloudBlobContainer container) {
+        for (ListBlobItem blobItem : container.listBlobs()) {
+            List<Pdf> pdfFiles = new ArrayList<>();
+            String zipFilename = FilenameUtils.getName(blobItem.getUri().toString());
+            byte[] metadataStream = null;
+
+            try {
+                BlobInputStream blobInputStream =
+                    container.getBlockBlobReference(zipFilename).openInputStream();
+
+                //Zip file will include metadata.json and collection of pdf documents
+                try (ZipInputStream zis = new ZipInputStream(blobInputStream)) {
+                    ZipEntry zipEntry;
+                    while ((zipEntry = zis.getNextEntry()) != null) {
+                        switch (FilenameUtils.getExtension(zipEntry.getName())) {
+                            case "json":
+                                metadataStream = toByteArray(zis);
+                                break;
```

Comment:
What if there are two json files in the zip file?
Should the second one overwrite `metadataStream`?

Reply:
Good point. Yes we will handle it as a separate story